mlflow/paddle/_paddle_autolog.py

```diff
@@ -0,0 +1,153 @@
+        self.metrics_logger = metrics_logger
+        self.run_id = run_id
+        self.log_models = log_models
+        self.log_every_n_epoch = log_every_n_epoch
+        self.epoch = 0
+
+    def _log_metrics(self, logs, current_epoch):
+        metrics = {
+            key: (metric[0] if isinstance(metric, list) else metric) for key, metric in logs.items()
+        }
+        self.metrics_logger.record_metrics(metrics, current_epoch)
+
+    def on_epoch_end(self, epoch, logs=None):
+        if self.model is not None and epoch % self.log_every_n_epoch == 0:
+            self._log_metrics(logs, epoch)
+            self.epoch = epoch
+
+    def on_train_begin(self, logs=None):
+        self.client.set_tags(self.run_id, {"Mode": "training"})
+        params = {
+            "optimizer_name": _get_optimizer_name(self.model._optimizer),
+            "learning_rate": self.model._optimizer._learning_rate,
+        }
+        self.client.log_params(self.run_id, params)
+        self.client.flush(synchronous=True)
+
+    def on_train_end(self, logs=None):
+        self.metrics_logger.flush()
+        self.client.flush(synchronous=True)
+
+    def on_eval_end(self, logs=None):
+        eval_logs = {
+            "eval_" + key: (metric[0] if isinstance(metric, list) else metric)
+            for key, metric in logs.items()
+        }
+        self._log_metrics(eval_logs, self.epoch)
+
+
+def _log_early_stop_params(early_stop_callback, client, run_id):
+    """
+    Logs early stopping configuration parameters to MLflow.
+    :param early_stop_callback: The early stopping callback instance used during training.
+    :param client: An `MlflowAutologgingQueueingClient` instance used for MLflow logging.
+    :param run_id: The ID of the MLflow Run to which to log configuration parameters.
+    """
+    client.log_params(
+        run_id,
+        {
+            p: getattr(early_stop_callback, p)
+            for p in ["monitor", "patience", "min_delta", "stopped_epoch"]
+            if hasattr(early_stop_callback, p)
+        },
+    )
+
+
+def _log_early_stop_metrics(early_stop_callback, client, run_id):
+    """
+    Logs early stopping behavior results (e.g. stopped epoch) as metrics to MLflow.
+    :param early_stop_callback: The early stopping callback instance used during training.
+    :param client: An `MlflowAutologgingQueueingClient` instance used for MLflow logging.
+    :param run_id: The ID of the MLflow Run to which to log configuration parameters.
+    """
+    if early_stop_callback.stopped_epoch == 0:
+        return
+
+    metrics = {
+        "stopped_epoch": early_stop_callback.stopped_epoch,
+        "restored_epoch": early_stop_callback.stopped_epoch - max(1, early_stop_callback.patience),
+        "best_value": early_stop_callback.best_value,
+        "wait_epoch": early_stop_callback.wait_epoch,
+    }
+
+    if early_stop_callback.best_weights is not None:
+        metrics["best_weights"] = early_stop_callback.best_weights
```

Comment:
We can only log a scala numeric value as a metric.

Reply:
got it